messaging-client-databus/src/test/java/com/inmobi/messaging/consumer/util/ConsumerUtil.java

```diff
@@ -416,6 +416,64 @@ public static void testTimeoutStats(ClientConfig config, String streamName,
         consumer.getMetrics())).getNumOfTiemOutsOnNext(), 10);
   }
 
+  public static void testDynamicCollector(ClientConfig config, String streamName,
+                                          String consumerName, boolean hadoop, Path[] rootDirs,
+                                          Configuration conf, String testStream,
+                                          String COLLECTOR_PREFIX) throws Exception {
+
+    AbstractMessagingDatabusConsumer consumer = createConsumer(hadoop);
+    consumer.init(streamName, consumerName, null, config);
+    Assert.assertEquals(consumer.getTopicName(), streamName);
+    Assert.assertEquals(consumer.getConsumerName(), consumerName);
+    Assert.assertEquals(consumer.getPartitionReaders().size(), 2);
+
+    //consume all the messages
+    int i;
+    for (i = 0; i < 300; i++) {
+      Message msg = consumer.next();
+      Assert.assertEquals(getMessage(msg.getData().array(), hadoop),
+          MessageUtil.constructMessage(i));
+    }
+    //add a collector
+    FileSystem fs = rootDirs[0].getFileSystem(conf);
+    Path collectorDir = new Path(rootDirs[0].toUri().toString(),
+        "data/" + testStream + "/" + COLLECTOR_PREFIX + "8") ;
+    fs.mkdirs(collectorDir);
+    String dataFile = TestUtil.files[2];
+    TestUtil.setUpCollectorDataFiles(fs, collectorDir, dataFile);
+
+    ((DatabusConsumer) consumer).createPartitionReaders();
```

Comment:
We can get rid of all these changes in test if we make partitionDiscovererAndReaderCreator timer task configurable.